ros2topic/ros2topic/api/__init__.py

```diff
@@ -37,6 +38,26 @@ def unsigned_int(string):
     return value
 
 
+def nonnegative_int(inval):
+    ret = int(inval)
+    if ret < 0:
+        # The error message here gets completely swallowed by argparse
+        raise ValueError('Value must be positive or zero')
+    return ret
+
+
+def positive_float(inval):
+    ret = float(inval)
```

Comment:
If `ValueError` formatting doesn't work well, catching that exception and reraising ArgumentTypeError would be better